src/snowflake/snowpark/modin/pandas/indexing.py

```diff
@@ -552,7 +552,7 @@ def _validate_locator_key(self, key: INDEXING_KEY_TYPE) -> None:
 
         if not (
             is_scalar(key)
-            or isinstance(key, (pd.Series, slice))
+            or isinstance(key, (pd.Series, slice, pd.Index))
             or is_list_like(key)
```

Comment:
add __iter__ so pd.Index should be is_list_like

Reply:
do you want me to add that to this PR or the next one? I was planning to add it to the next PR

Reply:
Let's add it in this PR so it can avoid some changes like this one

Reply:
Ok added